src/app/api/game-sessions/[gameSessionId]/download/route.ts

```diff
@@ -0,0 +1,60 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import { eq } from "drizzle-orm";
+import { z } from "zod";
+
+import { db } from "@/lib/db";
+import { bookingDetails, bookings, gameSessions, users } from "@/lib/db/schema";
+
+const routeContextSchema = z.object({
+  params: z.object({
+    gameSessionId: z.coerce.number(),
+  }),
+});
+
+export async function GET(
+  _req: NextRequest,
+  context: z.infer<typeof routeContextSchema>,
+) {
+  try {
+    const result = routeContextSchema.safeParse(context);
+    if (!result.success)
+      return new Response("Invalid id provided in the request", {
+        status: 400,
+      });
+
+    const {
+      params: { gameSessionId },
+    } = result.data;
+
+    const session = await db.query.gameSessions.findFirst({
+      where: eq(gameSessions.id, gameSessionId),
+    });
+
+    if (!session)
+      return new Response(`No Game Session found with id: ${gameSessionId}`, {
+        status: 404,
+      });
+    const [players] = await db
```

Comment:
No need for array destructing since we need everything, not just the first value